hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/security/x509/certificate/client/DefaultCertificateClient.java

```diff
@@ -1220,6 +1245,14 @@ public SCMSecurityProtocolClientSideTranslatorPB getScmSecureClient()
     return scmSecurityClient;
   }
 
+  public static void acquirePermit() throws InterruptedException {
+    semaphore.acquire();
```

Comment:
As we discussed, the SCMCertificateClient is the one that requires this semaphore for externally syncronizing the renewal, as that happens from within the SCM state machine in raft as part of the root CA rotation.

Some questions came to my mind regarding this:
Do we need the CertificateLifetimeMonitor within the SCMCertificateClient, if the whole thing is managed centrally via Ratis? Will a standalone SCM go through the procedure correctly, or for that we need some further adjustments in the logic?

Reply:
A SCM along doing the certificate rotation is not a common case.  I can just think one.  Here is the flow,
a.  SCM A bootstrapped with its sub CA root signed by root certficate 1.  Later some how it doesn't start and join the SCM HA immediately. 
b.  Root certificate was rotated from 1 to 2. 
c.  SCM A started and joined the SCM HA. 
d.  Some time later, SCM A found it need to rotate its certificate.

Reply:
It might happen together when the grace period is 0 seconds, so the two monitors are kicking in almost at the same time, besides that an other case, when there is a failure, and the wait ack is equal to the grace period, as then the first failure will push the second try to the time when the certificate monitor also kicks in.

These are rare cases, and should not cause any trouble due to the proper locking now, it might result in unnecessary renewals. I am not worried about those, I think it is just more explicit to have the certificate lifetime monitor explicitly disabled showing that it is not used in SCM. But let's leave ti for now as it is, I fine with it this way as well.